docs/programs/testing.md

```diff
@@ -0,0 +1,249 @@
+---
+title: "Testing with NodeJS"
+description: "Testing native solana programs written with rust using NodeJS"
+---
+
+When developing programs on Solana, ensuring their correctness and reliability
+is crucial. Until now devs have been using `solana-test-validator` for testing.
+This document covers testing your Solana program with Node.js  
+using `solana-bankrun`.
+
+## Overview
+
+There are two ways to test programs on Solana:
+
+1. [solana-test-validator](https://docs.solanalabs.com/cli/examples/test-validator):
+   That spins up a local emulator of the Solana Blockchain on your local machine
+   which receives the transactions to be processed by the validator.
+2. The various
+   [BanksClient-based](https://docs.rs/solana-banks-client/latest/solana_banks_client/)
+   test frameworks for SBF (Solana Bytecode Format) programs: Bankrun is a
+   framework that simulates a Solana bank’s operations, enabling developers to
+   deploy, interact with, and assess the behaviour of programs under test
+   conditions that mimic the mainnet. It helps set up the test environment and
+   offers tools for detailed transaction insights, enhancing debugging and
+   verification. With the client, we can load programs, and simulate and process
+   transactions seamlessly.
+   [solana-program-test](https://docs.rs/solana-program-test) (Rust),
+   [solana-bankrun](https://github.com/kevinheavey/solana-bankrun) (Rust,
+   JavaScript), [anchor-bankrun](https://www.npmjs.com/package/anchor-bankrun)
+   (Anchor, JavaScript),
+   [solders.bankrun](https://kevinheavey.github.io/solders/api_reference/bankrun.html)
+   (Python) are examples of the BanksClient-based testing framework.
+
+   ```
+   Note:
```

Comment:
@mikemaccana @heyAyushh is this code block intentional?

Reply:
Nope. Ayush tried to do a code block, I converted it to a quote and linked the docs but evidently left some backticks in (suggestions have their own backticks so it was probably that).  I'll fix it now.

https://github.com/solana-foundation/developer-content/pull/227#discussion_r1701978151

Reply:
https://github.com/solana-foundation/developer-content/pull/292

Reply:
Thanks @mikemaccana, sorry it slipped